src/core/component.ts

```diff
@@ -71,7 +71,35 @@ const BaseComponent: VueConstructor<IBaseComponent> = Vue.extend({
         );
     },
 
+    beforeUpdate() {
+        this.$_config.setPrevNestedOptions(this.$_config.getNestedOptionValues());
+    },
+
     updated() {
+        if (this.$_config.hasOptionsToUpdate) {
+            const options = this.$_config.getNestedOptionValues();
+            const nestedOptions = Object.keys(options);
+            const prevNestedOptions = Object.keys(this.$_config.prevNestedOptions);
+            if (nestedOptions.length < prevNestedOptions.length) {
```

Comment:
If an item is replaced with another item, the update is going to be skipped, I think.
You could modify the test above:
```diff
    it("replaces nested component by condition", (done) => {
        const vm = new Vue({
            template:
                `<test-component>` +
                `  <nested v-if="showNest" :prop1="value" />` +
+               `  <nested v-if="!showNest" :prop1="456" />` +
                `  <nested :prop1="321" />` +
                `</test-component>`,

```

Reply:
It works on my side